util/get_radar_data.py

```diff
@@ -0,0 +1,76 @@
+import csv
+import coinach
+import os
+from collections import defaultdict
+
+_OUTPUT_FILE = 'hunt.js'
+
+
+def parse_data(csvfile, lang='en'):
+    monsters = {}
+    reader = csv.reader(csvfile)
+    # skip the first three header lines
+    next(reader)
+    next(reader)
+    next(reader)
+    if lang == "chs":
+        lang = "cn"
+    for row in reader:
+        print(row)
+        m_id, m_base, m_rank, m_name = row[:-1]
+        if not m_name: continue
+        m_rank = 'S' if m_rank == 3 else (
+            'A' if m_rank == 2 else 'B'
+        )
+        if m_base == 'BNpcBase#10422':
+            m_rank = 'SS+'
+        if m_base == 'BNpcBase#10755':
+            m_rank = 'SS-'
+        monsters[m_id] = {
+            'name':{
+                lang: m_name,
+            },
+            'rank': m_rank,
+        }
+    return monsters
+
+
+def update(reader, writer):
+    languages = ['en', 'de', 'fr', 'ja']
+    monsters = defaultdict(lambda: {'name':{}, 'rank':''})
+    for locale in languages:
+        data = reader.exd('NotoriousMonster', lang=locale)
+        data = parse_data(data, lang=locale)
+        for m in data.keys():
+            monsters[m]['name'].update(data[m]['name'])
+            monsters[m]['rank'] = data[m]['rank']
+    return monsters
+
+def update_cn(reader, writer, monsters={}):
+    languages = ['chs']
+    for locale in languages:
+        data = reader.exd('NotoriousMonster', lang=locale)
+        data = parse_data(data, lang=locale)
+        for m in data.keys():
+            monsters[m]['name'].update(data[m]['name'])
+            monsters[m]['rank'] = data[m]['rank']
+    return monsters
+
+
+CN_FFXIV_PATH = 'C:\Games\SDO\FFXIV'
```

Comment:
Do I need the chinese version to get the chinese names from Saint Coinach?

Reply:
You don't need to modify SaintCoinach to support Chinese client, it works well to export the files.
But yes, you need a Chinese version FFXIV client and use SaintCoinach to export.
We also have a git repo to do the data mining of CN client [ffxiv-datamining-cn](https://github.com/thewakingsands/ffxiv-datamining-cn).

Reply:
Hmmmmmmmmm.  How hard would it be to use xivapi and the chinese version of that instead of requiring keeping a second ffxiv client up to date on my computer?

Reply:
As you said before, xivapi has not yet hosted this csv file?
Or do I misunderstand what you mean?

Reply:
Maybe we're misunderstanding each other.  I was asking how hard it would be to use xivapi instead of saint coinach here.  That way updating the radar wouldn't involve installing multiple versions of the client.

It looks like it has these values: https://xivapi.com/NotoriousMonster/22?pretty=true

Reply:
lmao I didn’t use xivapi much so I thought your “xivapi doesn’t like to use it” means this csv is not hosted in xivapi.
I’ll change the script to a xivapi version tomorrow.